### PR TITLE
Default tag ordering to the primary key

### PR DESCRIPTION
**Pull request by rtpg**
This is based off of work done by @steverecio on #871 

This doesn't cause a change in query count.

@steverecio would you mind looking over this? And do you want to be placed in the `AUTHORS` file? I appreciate you digging through this, your PR was very helpful and helped me to understand what needed to be done to get this working.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -41,7 +41,7 @@ exclude = tests*
 [flake8]
 # E501: line too long
 ignore = E501
-exclude = .venv,.git,.tox
+exclude = .venv,.git,.tox,.direnv
```
Comment (rtpg) [inline setup.cfg:44]: been meaning to add this for a while....

File: taggit/managers.py [hunk 2]
```diff
@@ -206,59 +215,82 @@ def add(self, *tags, through_defaults=None, tag_kwargs=None, **kwargs):
...
+            # we're going to end up doing this query anyways, so here is fine
+            for t in existing:
+                existing_tags_for_str[t.name] = t
+
+        result = []
+        # this set is used for deduplicating tags
+        seen_tags = set()
+        for t in tags:
+            if isinstance(t, self.through.tag_model()):
+                if t in seen_tags:
```
Comment (steverecio) [inline taggit/managers.py:258]: nitpicky but could be simpler here to do:

```
if not t in seen_tags:
    seen_tags.add(t)
    result.append(t)
```

File: taggit/managers.py [hunk 3]
```diff
@@ -206,59 +215,82 @@ def add(self, *tags, through_defaults=None, tag_kwargs=None, **kwargs):
...
+                        lookup = {"name": t, **tag_kwargs}
+                    existing_tag, _ = manager.get_or_create(
+                        **lookup, defaults={"name": t}
+                    )
+                # we now have an existing tag for this string
 
-        tag_objs.update(existing)
+                # confirm if we've seen it or not (this is where case insensitivity comes
+                # into play)
+                if existing_tag in seen_tags:
```
Comment (steverecio) [inline taggit/managers.py:281]: likewise we can eliminate the continue here

**Comment by steverecio** [comment]
@rtpg thanks for putting this together! I left two minor comments but I think it looks great and is a cleaner refactor! Would love to see this merged in. I appreciate the offer! Would love to be included on the `AUTHORS` file. You can use "Steve Recio <stevenrecio@gmail.com>" :)

File: AUTHORS [hunk 4]
```diff
@@ -19,3 +19,4 @@ Andrew Pryde <andrew@rocketpod.co.uk>
 John Whitlock <jwhitlock@mozilla.com>
 Jon Dufresne <jon.dufresne@gmail.com>
 Pablo Olmedo Dorado <pablolmedorado@gmail.com>
+Steve Reciao <stevenrecio@gmail.com>
```
Comment (steverecio) [inline AUTHORS:22]: *Recio 😎

**Pull request merged**